docs/tutorials/algokit-template.md

```diff
@@ -0,0 +1,141 @@
+
+### Bootstrap Option
+
+AlgoKit templates can include a bootstrap script. This script is automatically run after the project is initialized and can perform various setup tasks like installing dependencies or setting up databases.
+
+- `env`: Copies an `.env.template` file to `.env` in the current working directory and prompts for any unspecified values. This is a critical aspect of managing environment variables securely, ensuring that sensitive data doesn't accidentally end up in version control.
+- `poetry`: If your Python project uses Poetry for dependency management, the `poetry` command installs Poetry (if not present) and runs `poetry install` in the current working directory to install Python dependencies.
+- `npm`: If you're developing a JavaScript or TypeScript project, the `npm` command runs npm install in the current working directory to install Node.js dependencies.
+- `all`: The `all` command runs all the aforementioned bootstrap sub-commands in the current directory and its subdirectories. This command is a comprehensive way to ensure all project dependencies and environment variables are properly set up.
+
+### Predefined copier answers
+
+When initializing a new project, Copier can prompt the user for input, which is then passed to the template as variables. This is useful for customizing the new project based on user input.
+
+Example:
+
+```yaml
+# copier.yaml
+project_name:
+    type: str
+    help: What is the name of this project.
+    placeholder: "algorand-app"
+```
+This would prompt the user for the project name, and the input can then be used in the template using the Jinja syntax `{{ project_name }}`.
+
+#### Default Behaviors
+When creating an AlgoKit template, there are a few default behaviors that you can include to improve the development experience:
+
+- **Git**: If Git is installed on the user's system and the user's working directory is a Git repository, Copier will commit the newly created project as a new commit in the repository. This feature helps to maintain a clean version history for the project. If you wish to add a specific commit message for this action, you can specify a `commit_message` in the `_commit` option in your `copier.yaml` file.
+
+- **VSCode**: If the user has Visual Studio Code (VSCode) installed and the path to VSCode is added to their system's PATH, Copier can automatically open the newly created project in a new VSCode window. This is triggered by adding `vscode: true` in your `copier.yam` file.
+
+- **Bootstrap**: As previously mentioned, AlgoKit templates can include a bootstrap script that is automatically run after the project is initialized. This script can perform various setup tasks like installing dependencies or setting up databases. By default, if a `bootstrap` task is defined in the `copier.yaml`, it will be executed unless the user opts out during the prompt.
```

Comment:
Similar to previous comments, bootsrap is handled by algokit cli its not controlled from within the codebase of the template that user creates, please sligtly rephrase to make it clear who is executing the bootstrap